pkg/controller/subjectpermission/subjectpermission_controller.go

```diff
@@ -171,18 +171,91 @@ func (r *ReconcileSubjectPermission) Reconcile(request reconcile.Request) (recon
 		return reconcile.Result{}, nil
 	}
 
+	// get the Namespace instance
+	ns := &corev1.Namespace{}
+	err = r.client.Get(context.TODO(), request.NamespacedName, ns)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Object not found, it can be transitioning to the final desired state
+			// e. g. deletion or creation still in progress. Return and retry again
+			reqLogger.Info("Object not ready")
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object
+		reqLogger.Info("Error Getting Namespace")
+		return reconcile.Result{}, err
+	}
+
+	// get the NamespaceList
+	nsList := &corev1.NamespaceList{}
+	opts = client.ListOptions{Namespace: request.Namespace}
+	err = r.client.List(context.TODO(), &opts, nsList)
+	if err != nil {
+		reqLogger.Error(err, "Failed to get clusterRoleBindingList")
+		return reconcile.Result{}, err
+	}
+
+	// slice of clusterRoleName that does not exists as a clusterRole
+	permissionClusterRoleNameList := controllerutil.PopulateCrPermissionClusterRoleNames(instance, clusterRoleList)
+	for _, permissionClusterRoleName := range permissionClusterRoleNameList {
+		// update condition
+		updatedSubjectPermission := controllerutil.UpdateCondition(instance, permissionClusterRoleName+" for clusterPermission does not exist", permissionClusterRoleName, true, managedv1alpha1.SubjectPermissionFailed)
+		err = r.client.Status().Update(context.TODO(), updatedSubjectPermission)
+		if err != nil {
+			reqLogger.Error(err, "Failed to update condition.")
+			return reconcile.Result{}, err
+		}
```

Comment:
We need to return after each Update() call... not sure this is how conditions should work (but I know its in the spec).

Reply:
I am as well. @lisa @jharrington22 can I get your opinion on this?

Reply:
@fahlmant Could you say a bit more about what you mean on this one?

Reply:
After each update its going to trigger another reconcile loop iteration, as a general rule when you update an object we want to return from the reconcile loop and allow the next iteration to continue.

Reply:
This helps make reconcile loop iteration idempotent and safe

Reply:
had a discussion with @jharrington22 . Follow up on conversation in design doc: https://docs.google.com/document/d/1eTy4i2M-ay5bQ9FzJ1YS3MxBFuzhJe3WbREexVmfROI/edit?ts=5d8123a2#heading=h.7f38tn424ok4

Reply:
had a discussion with @jharrington22 . Follow up on conversation in design doc comment